<launch>
  <node name="map_server" pkg="map_server" type="map_server" args="$(find collvoid_stage)/world/swarmlab_map.yaml"/>
  <rosparam command="load" file="$(find collvoid_stage)/params_created.yaml"/>
  <node pkg="stage_ros" type="stageros" name="stageros" args="$(find collvoid_stage)/world/swarmlab_created.world" respawn="false" output="screen" />
  <node name="fake_localization" pkg="fake_localization" ns="robot_0" type="fake_localization" respawn="false">
    <param name="~tf_prefix" value="robot_0" />
    <param name="~odom_frame_id" value="/robot_0/odom" />
    <param name="~base_frame_id" value="/robot_0/base_link" />
   </node>
   <!--  <node pkg="move_base" type="move_base" respawn="true" name="move_base" ns="robot_0" output="screen"> -->
  <!--   <rosparam command="load" file="$(find collvoid_stage)/params/params_pr2.yaml" /> -->
  <!--   <rosparam command="load" file="$(find collvoid_stage)/params/collvoid_config.yaml" /> -->
  <!--   <remap from="map" to="/map" /> -->
  <!--   <param name="~tf_prefix" value="robot_0" /> -->
  <!--   <param name="~controlled" value="true" /> -->

  <!--   <param name="~/global_costmap/robot_base_frame" value="robot_0/base_link" />  -->
  <!--   <param name="~/local_costmap/robot_base_frame" value="robot_0/base_link" />  -->
  <!--   <param name="~/local_costmap/global_frame" value="robot_0/odom" />  -->
  <!--   <param name="base_local_planner" value="collvoid_local_planner/CollvoidLocalPlanner" /> -->
  <!--   <param name="base_global_planner" value="collvoid_simple_global_planner/CollvoidSimpleGlobalPlanner" /> -->
  <!--   <remap from="/position_share_in" to="/position_share" /> -->
  <!--   <remap from="/position_share_out" to="/position_share" /> -->
  <!-- </node>  -->
  <node pkg="collvoid_controller" type="controller_robots.py" name="controller_robots" ns="robot_0" output="screen" />

  <node name="fake_localization" pkg="fake_localization" ns="robot_1" type="fake_localization" respawn="false">
    <param name="~tf_prefix" value="robot_1" />
    <param name="~odom_frame_id" value="/robot_1/odom" />
    <param name="~base_frame_id" value="/robot_1/base_link" />
  </node>

  <node pkg="collvoid_controller" type="active_collision_avoidance.py" name="active_collvoid" ns="robot_0" output="screen">
    <param name="~base_frame_id" value="/robot_0/base_link" />
  </node>
  <node pkg="move_base" type="move_base" respawn="true" name="move_base" ns="robot_1" output="screen">
    <rosparam command="load" file="$(find collvoid_stage)/params/params_pr2.yaml" />
    <rosparam command="load" file="$(find collvoid_stage)/params/collvoid_config.yaml" />
    <remap from="map" to="/map" />
    <param name="~controlled" value="false" />

    <param name="~tf_prefix" value="robot_1" />
    <param name="~/global_costmap/robot_base_frame" value="robot_1/base_link" /> 
    <param name="~/local_costmap/robot_base_frame" value="robot_1/base_link" /> 
    <param name="~/local_costmap/global_frame" value="robot_1/odom" /> 
    <param name="base_local_planner" value="collvoid_local_planner/CollvoidLocalPlanner" />
    <param name="base_global_planner" value="collvoid_simple_global_planner/CollvoidSimpleGlobalPlanner" />
    <remap from="/position_share_in" to="/position_share" />
    <remap from="/position_share_out" to="/position_share" />
  </node> 
  <node pkg="collvoid_controller" type="controller_robots.py" name="controller_robots" ns="robot_1" output="screen" />
  <node pkg="collvoid_controller" type="controller.py" name="controller" output="screen"/>
  <node pkg="rviz" type="rviz" name="rviz" args="-d $(find collvoid_stage)/multi_view.rviz" output="screen" />
</launch>
